.github/workflows/release.yml

```diff
@@ -1,28 +1,126 @@
 name: Release
 
 on:
-  push:
-    tags:
-      - 'v[0-9]+.[0-9]+.[0-9]+'
+  workflow_dispatch:
+    inputs:
+      releaseType:
+        type: choice
+        options:
+          - minor
+          - patch
+        default: minor
+        required: true
+        description: 'minor: v0.X.0, patch: v0.0.X'
+      debug:
+        type: boolean
+        default: true
+        description: 'executes the workflow in debug mode (skip the publishing tag, docker image and release steps)'
 
 jobs:
-  release:
-    name: Create Release
-    if: github.event.base_ref == 'refs/heads/master'
-    runs-on: ubuntu-20.04
+  check-permission:
+    name: Check permission
+    if: contains(github.ref, 'refs/heads/master')
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
     steps:
-      - name: Get Version
-        id: get_version
+      - name: Check user permission
+        uses: actions-cool/check-user-permission@v2.2.0
+        id: check
+        with:
+          require: 'write'
+    outputs:
+      hasWritePermission: ${{ steps.check.outputs.require-result }}
+
+  publish-tag:
+    name: Publish tag
+    needs: check-permission
+    if: contains(needs.check-permission.outputs.hasWritePermission, 'true')
```

Comment:
Could we just do if: needs.check-permission.outputs.hasWritePermission ?

Reply:
`hasWritePermission` is exported into env by `check-permission` steps. When bool are exported they get converted into string (check https://github.com/actions/runner/issues/1483). So for string comparison `contains(needs.check-permission.outputs.hasWritePermission, 'true')` is used. This is equivalent of `needs.check-permission.outputs.hasWritePermission == "true"`